core/enclave.go

```diff
@@ -2,26 +2,23 @@ package core
 
 import (
 	"errors"
-	"sync/atomic"
-	"unsafe"
+	"sync"
 )
 
 var (
-	key unsafe.Pointer // *Coffer
+	key    = &Coffer{}
+	keyMtx = sync.Mutex{}
 )
 
-func init() {
-	if swapped := atomic.CompareAndSwapPointer(&key, nil, nil); swapped {
-		setKey(NewCoffer())
-	}
-}
+func getKey(create bool) *Coffer {
```

Comment:
IMO a separate `createKey` function would be more speaking instead of doing flow-control using the parameter... But that might be a matter of taste...